### PR TITLE
Update counter_face.c: Non-blocking sound, silent option

### PR DESCRIPTION
**Pull request by Hein-NonesensE**
Hi, I have changed the sound to be non-blocking, so we can count faster than the sound is played. And I added an option to deactivate sound, for those of us who prefer silence.

I hope you like it and I hope it's up to standard as I'm not a pro(grammer). At least, it works on my watch :-)

### CODE REVIEW

File: movement/watch_faces/complication/counter_face.c [hunk 1]
```diff
@@ -48,12 +54,25 @@ bool counter_face_loop(movement_event_t event, movement_settings_t *settings, vo
 
     switch (event.event_type) {
         case EVENT_ALARM_BUTTON_UP:
+            watch_buzzer_abort_sequence; //abort running buzzer sequence when counting fast
```
Comment (neutralinsomniac) [inline movement/watch_faces/complication/counter_face.c:57]: This call and the other call on line 68 need to be `watch_buzzer_abort_sequence();`

Without the parentheses you aren’t actually calling a function.

File: movement/watch_faces/complication/counter_face.c [hunk 2]
```diff
@@ -75,22 +94,46 @@ bool counter_face_loop(movement_event_t event, movement_settings_t *settings, vo
...
+        i++;
+        sound_seq[i] = 6;
+        i++;
+    }
+    if (high_count > 1) {
+        sound_seq[i] = -2;
+        i++;
+        sound_seq[i] = high_count-1;
+    }
+    watch_buzzer_play_sequence((int8_t *)sound_seq, NULL);
```
Comment (neutralinsomniac) [inline movement/watch_faces/complication/counter_face.c:136]: sound_seq is a stack variable, and this function goes out of scope as soon as `watch_buzzer_play_sequence()` returns. This means that while the audio is playing (asynchronously), sound_seq can be overwritten by any future function calls pushing things onto the stack. `sound_seq` should probably be static, but also keep in mind then that your initialization will only take effect on the first call of `beep_counter()` so you may want to `memset()` it.

Comment (Hein-NonesensE) [reply movement/watch_faces/complication/counter_face.c:136]: Thanks for your feedback! I implemented the suggested changes in a new commit.

File: movement/watch_faces/complication/counter_face.c [hunk 3]
```diff
@@ -38,6 +40,9 @@ void counter_face_setup(movement_settings_t *settings, uint8_t watch_face_index,
 
 void counter_face_activate(movement_settings_t *settings, void *context) {
     (void) settings;
+    if (beep_on){
```
Comment (Kistelini) [inline movement/watch_faces/complication/counter_face.c:43]:
```suggestion
    if (beep_on) {
```

File: movement/watch_faces/complication/counter_face.c [hunk 4]
```diff
@@ -48,12 +53,25 @@ bool counter_face_loop(movement_event_t event, movement_settings_t *settings, vo
...
     switch (event.event_type) {
         case EVENT_ALARM_BUTTON_UP:
+            watch_buzzer_abort_sequence(); //abort running buzzer sequence when counting fast
             state->counter_idx++; // increment counter index
             if (state->counter_idx>99) { //0-99
                 state->counter_idx=0;//reset counter index
             }
             print_counter(state);
+            if (beep_on) {
             beep_counter(state);
```
Comment (Kistelini) [inline movement/watch_faces/complication/counter_face.c:63]:
```suggestion
                beep_counter(state);
```

File: movement/watch_faces/complication/counter_face.c [hunk 5]
```diff
@@ -48,12 +53,25 @@ bool counter_face_loop(movement_event_t event, movement_settings_t *settings, vo
...
+            }
+            break;
+        case EVENT_LIGHT_LONG_PRESS:
+            watch_buzzer_abort_sequence(); 
+            beep_on = !beep_on;
+            if (beep_on){
+            watch_set_indicator(WATCH_INDICATOR_SIGNAL);
+            }
+            else {
+            watch_clear_indicator(WATCH_INDICATOR_SIGNAL);
```
Comment (Kistelini) [inline movement/watch_faces/complication/counter_face.c:73]:
```suggestion
            if (beep_on) {
                watch_set_indicator(WATCH_INDICATOR_SIGNAL);
            } else {
                watch_clear_indicator(WATCH_INDICATOR_SIGNAL);
```

File: movement/watch_faces/complication/counter_face.c [hunk 6]
```diff
@@ -75,22 +93,47 @@ bool counter_face_loop(movement_event_t event, movement_settings_t *settings, vo
 
 // beep counter index times
 void beep_counter(counter_state_t *state) {
-	
 	int low_count = state->counter_idx/5;
```
Comment (Kistelini) [inline movement/watch_faces/complication/counter_face.c:96]: Indentation here was already mixed, maybe you can fix it, too?

**Pull request closed without merging**